### DISCUSSION TITLE
It would be nice if magit displayed the date of a stash

### DISCUSSION THREAD

**Discussion by steinarb** [magit/magit#5200] [Support] [answered]
Today I wanted to find out when I had created a stash.

magit didn't have anything looking like a date, so I googled and found this (which worked) https://stackoverflow.com/a/15551690

It would be nice if magit could use this and extract the date and display the date somewhere on the stash (it's ok to have it hidden by default as long as it is accessible with a few key clicks).

**Comment by tarsius** [comment]
The date is displayed in the optional margin, which you can toggle on the fly using `L L`. To display the margin by default, customize the options in the `magit-margin` group.

**Reply by steinarb** [reply]
Doesn't work for me in 20240811.1419 from melpa on emacs 29.4, unfortunately.

When i press ```L``` in a magit buffer, the minibuffer expands with alternatives (same as in rebase) and ```L ``` is one of the alternatives (Margin "Toggle visibility", which sounds right).

But nothing happens when I press ```L```.

If I enter the stash itself and press  ```L L``` I'm told "Magit margin isn't supported in this buffer"

**Reply by tarsius** [reply]
> But nothing happens when I press `L`.

![20240814-20:03:27](https://github.com/user-attachments/assets/001d9e7b-ffff-4bb3-b6af-4ad89f975cab)

does not appear?

**Reply by steinarb** [reply]
No it is stuck like this after I press ```L```, nothing more happens no matter what key I press.
![Screenshot at 2024-08-14 20-57-49](https://github.com/user-attachments/assets/206ed64e-65c9-4b30-bad1-70d9260c709d)

**Reply by steinarb** [reply]
Or rather: ```g``` and ```s``` and ```w``` close the popup without any clear indication of what they do.

But ```L``` or ```l``` or ```d``` or ```x``` does nothing that I can see.

**Reply by tarsius** [reply]
The

```
2m
2m
2m
2y
```

is a more compact form of that very same margin. "2m" for example means "2 months [ago]". You can change the amount of information being displayed in the margin by using the other commands under the "Margin" group in the menu (such as `l Cycle style`).

**Reply by steinarb** [reply]
Heh... I didn't understand what you were talking about until I suddenly saw it in my own screenshot.

I didn't see those appear. But now I can see them appear and disappear when I press ```L```.

But pressing ```l``` once got me here, but pressing ```l``` again doesn't seem to change anything
![Screenshot at 2024-08-14 21-25-50](https://github.com/user-attachments/assets/0c9c1614-ff9c-4829-b717-adeedcbf3249)

**Reply by tarsius** [reply, marked as answer]
Somewhere along the way you pressed `x`. Until you the `x` the again you are in "show shortstat instead of date and maybe author" mode. The style and detail commands only affect the "show date and maybe author" mode.

**Reply by steinarb** [reply]
Ah! This looks like what I was looking for. Thanks!
![Screenshot at 2024-08-14 22-01-47](https://github.com/user-attachments/assets/10bc5e19-d174-4b3e-8def-d770b844757e)

**Reply by tarsius** [reply]
I'm glad we've got it figured out.
